lib/Conversion/HandshakeToFIRRTL/HandshakeToFIRRTL.cpp

```diff
@@ -254,14 +265,50 @@ static Value createConstantOp(FIRRTLType opType, APInt value,
   return Value();
 }
 
-static Type getHandshakeBundleDataType(BundleType bundle) {
-  if (auto dataType = bundle.getElementType("data")) {
-    auto intType = dataType.cast<firrtl::IntType>();
-    return IntegerType::get(bundle.getContext(), intType.getWidthOrSentinel(),
+/// Creates a Value that has an assigned zero value. For bundles, this
+/// corresponds to assigning zero to each element recursively.
+static Value createZeroDataConst(FIRRTLType dataType, Location insertLoc,
+                                 OpBuilder &builder) {
+  return TypeSwitch<Type, Value>(dataType)
+      .Case<IntType>([&](IntType dataType) {
```

Comment:
Use `.Case<IntType>([&](auto dataType)` for DRYness.

Reply:
I guess this file would benefit from some cleanup work. There are a many parts that could be improved with such changes.

Reply:
Indeed... However, we've been saying (and hoping) for a while now that no more effort should go into HandshakeToFIRRTL, and instead, HandshakeToHW should be properly elaborated... However, that effort has never fully gotten off the ground.